README.rst

```diff
@@ -939,6 +960,78 @@ Once you have performed the 3 steps above, you can test it as follows:
 
+How to configure automatic provisioning of subnets and IPs
+----------------------------------------------------------
+
+The following steps will help you configure automatic provisioning of subnets and IPs
+for devices:
+
+1. Create a Subnet and a Subnet Division Rule
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create a master subnet under which automatically generated subnets will be provisioned.
+
+**Note**: Choose the size of the subnet appropriately considering your use case.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/issues/400-subnet-subdivision-rule/docs/subnet-division-rule/subnet.png
+  :alt: Creating a master subnet example
+
+On the same page, add a **subnet division rule** that will be used to provision subnets
+under the master subnet.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/issues/400-subnet-subdivision-rule/docs/subnet-division-rule/subnet-division-rule.png
+  :alt: Creating a subnet division rule example
+
+2. Create a VPN Server
+~~~~~~~~~~~~~~~~~~~~~~
+
+Now create a VPN Server and choose the previously created **master subnet** as the subnet for
+this VPN Server.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/issues/400-subnet-subdivision-rule/docs/subnet-division-rule/vpn-server.png
+  :alt: Creating a VPN Server example
+
+3. Create a VPN Client Template
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create a template, setting the **Type** field to **VPN Client** and **VPN** field to use the
+previously created VPN Server.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/issues/400-subnet-subdivision-rule/docs/subnet-division-rule/vpn-client.png
+  :alt: Creating a VPN Client template example
+
+**Note**: You can also check the **Enable by default** field if you want to automatically
+apply this template to devices that will register in future.
+
+4. Apply VPN Client Template to Devices
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+With everything in place, you can now apply the VPN Client Template to devices.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/issues/400-subnet-subdivision-rule/docs/subnet-division-rule/apply-template-to-device.png
+  :alt: Adding template to device example
+
+After saving the device, you should see all provisioned Subnets and IPs for this device
+under the **System Defined Variables**.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/issues/400-subnet-subdivision-rule/docs/subnet-division-rule/subnet-division-rule.png
+  :alt: Provisioned Subnets and IPs available as System Defined Variables example
+
+Voila! You can now use these variables in configuration of the device. Refer to `How to use configuration variables <#how-to-use-configuration-variables>`_
+section of this documentation to learn how to use configuration variables.
+
```

Comment:
can you please show that the created subnets and ip addresses are available as system defined variables in the device?

Eg:

- show a screenshot in which the system defined variables are visible

Then mentions these variables can be used in templates, provided the default fall back values are provided in the "default values" template field (used mainly for validation).